dangerfile.js

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-console */
+// inspire by reacts dangerfile
+// danger has to be the first thing required!
+const { danger, markdown } = require('danger');
+const { exec } = require('child_process');
+const { loadComparison } = require('./scripts/sizeSnapshot');
+
+const parsedSizeChangeThreshold = 300;
+const gzipSizeChangeThreshold = 100;
+
+/**
+ * executes a git subcommand
+ * @param {any} args
+ */
+function git(args) {
+  return new Promise((resolve, reject) => {
+    exec(`git ${args}`, (err, stdout) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(stdout.trim());
+      }
+    });
+  });
+}
+
+const upstreamRemote = 'danger-upstream';
+
+async function cleanup() {
+  await git(`remote remove ${upstreamRemote}`);
+}
+
+/**
+ * creates a callback for Object.entries(comparison).filter that excludes every
+ * entry that does not exceed the given threshold values for parsed and gzip size
+ *
+ * @param {number} parsedThreshold
+ * @param {number} gzipThreshold
+ */
+function createComparisonFilter(parsedThreshold, gzipThreshold) {
+  return ([, { parsed, gzip }]) => {
+    return (
+      Math.abs(parsed.absoluteDiff) >= parsedThreshold ||
+      Math.abs(gzip.absoluteDiff) >= gzipThreshold
+    );
+  };
+}
+
+/**
+ * checks if the bundle is of a package e.b. `@material-ui/core` but not
+ * `@material-ui/core/Paper`
+ * @param {[string]} comparison entry
+ */
+function isPackageComparison([bundle]) {
+  return /^@[\w-]+\/[\w-]+$/.test(bundle);
+}
+
+/**
+ * Generates a user-readable string from a percentage change
+ * @param {number} change
+ * @param {string} goodEmoji emoji on reduction
+ * @param {string} badEmooji emoji on increase
+ */
+function addPercent(change, goodEmoji = '', badEmooji = ':small_red_triangle:') {
+  const formatted = (change * 100).toFixed(2);
+  if (/^-|^0(?:\.0+)$/.test(formatted)) {
+    return `${formatted}%${goodEmoji}`;
+  }
+  return `+${formatted}%${badEmooji}`;
+}
+
+/**
+ * Generates a Markdown table
+ * @param {string[]} headers
+ * @param {string[][]} body
+ * @returns {string}
+ */
+function generateMDTable(headers, body) {
```

Comment:
Note for self: we could have used this abstraction the API markdown generation logic.